client/client.go

```diff
@@ -95,6 +108,35 @@ func (c *Client) Do(ctx context.Context, pctx *plancontext.Context, fn DoFunc) e
 	return eg.Wait()
 }
 
+// detectPlatform will try to autoatically Buildkit's target platform
+// automatically. If not possible, default platform will be used.
+func (c *Client) detectPlatform(ctx context.Context) (*specs.Platform, error) {
+	w, err := c.c.ListWorkers(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error detecting platform %w", err)
+	}
+
+	lg := log.Ctx(ctx)
+	// TODO what do we do when we have multiple workers?
```

Comment:
@TomChv @aluzzardi do you have any opinions here? 

Buildkit could potentially return many workers, but we're selecting the first one as it's the default returned by buildkit (https://github.com/moby/buildkit/blob/master/worker/workercontroller.go#L59-L71). 

Is there anything else we should do here initially?

Reply:
Sorry, didn't saw the update, here's a duplicate from my commit below https://github.com/dagger/dagger/pull/2028#discussion_r841956107

> As I said in our private discussion, we should first ignore all platforms that do not match host's OS.
After that I think we should make a benchmark in our side to see which platforms is the faster when we it's not host's one and then we choose it.

Reply:
> @TomChv @aluzzardi do you have any opinions here?

Great PR, @marcosnils!

docker build (and especially https://github.com/docker/buildx) is doing the same thing -- I would look into their implementation and just mimic the behavior (or ask on Slack -- I quickly glanced over the code and it's not immediately obvious how they pick the default platform).